### PR TITLE
fix(prelude): remove widgets module from prelude

### PR DESCRIPTION
**Pull request by joshka**
This helps to keep the prelude small and less likely to conflict with
other crates.

- remove widgets module from prelude as the entire module can be just as
  easily imported with `use ratatui::widgets::*;`
- move prelude module into its own file
- update examples to import widgets module instead of just prelude
- added several modules to prelude to make it possible to qualify
  imports that collide with other types that have similar names

## Note

In the [discussion of this on discord](https://discord.com/channels/1070692720437383208/1072879985762762812/1129381084010909756), just the canvas was suggested to be removed, however given that it's fairly easy to import widgets::*, it makes sense to drop the entire module from the prelude.

### CODE REVIEW

**Review by sayanarijit — APPROVED** [review]

**Comment by joshka** [comment]
Will wait on a second review on this one rather than merging.

**Review by mindoodoo — APPROVED** [review]

**Review by orhun — APPROVED** [review]

**Pull request merged**
